File: app/src/main/java/com/marknkamau/justjava/activities/checkout/CheckoutActivityView.java
package com.marknkamau.justjava.activities.checkout;

import com.google.firebase.auth.FirebaseUser;

import java.util.Map;

public interface CheckoutActivityView {
    void setLoggedInStatus(Boolean status);
    void invalidateMenu();
    void setDisplayToLoggedIn(FirebaseUser user, Map<String, String> defaults);
    void setDisplayToLoggedOut();
    void showUploadBar();
    void hideUploadBar();
    void finishActivity();
    void showMessage(String message);
}

File: app/src/main/java/com/marknkamau/justjava/ui/checkout/CheckoutActivityView.kt
package com.marknkamau.justjava.ui.checkout

import com.google.firebase.auth.FirebaseUser

interface CheckoutActivityView {
    fun setLoggedInStatus(status: Boolean?)
    fun invalidateMenu()
    fun setDisplayToLoggedIn(user: FirebaseUser, defaults: Map<String, String>)
    fun setDisplayToLoggedOut()
    fun showUploadBar()
    fun hideUploadBar()
    fun finishActivity()
    fun showMessage(message: String?)
}
